### ISSUE TITLE
UMD modules don't augment correctly

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#9343]
```ts
// a.d.ts
export as namespace MyFoo;
export interface Foo {
  a: string;
}

// b.d.ts
import * as A from './a';
declare module './a' {
  interface Foo {
    b: string;
  }
}

// c.ts
/// <reference path="b.d.ts" />

import * as MyFoo from './a';

var p: MyFoo.Foo;
console.log(p.a);
console.log(p.b);
```

> tsc c.ts

Actual:
```
c.ts(3,14): error TS2305: Module '"C:/Throwaway/umdaug/a".MyFoo' has no exported member 'Foo'.
```

Expected: No error

**Comment by RyanCavanaugh** [comment]
Oops not repro
